lib/improver/cli/extract.py

```diff
@@ -96,27 +53,43 @@ def process(cube, constraints, units=None):
             The Cube from which a sub-cube is extracted
         constraints (list):
             The constraint(s) to be applied.  These must be of the form
-            "key=value", eg "threshold=1".  Scalars, boolean and string
-            values are supported.  Comma-separated lists
-            e.g. key=[value1,value2,value3]
-            are supported.
-            These comma-separated lists can either extract all values
-            specified in the list or all values specified within a range
-            e.g. key=[value1:value3].
+            "key=value", eg "threshold=1". Multiple constraints can be
+            specified using a comma separated list (no spaces), e.g.
+            threshold=1,wind_speed=10. Scalars, boolean and string
+            values are supported. Lists of values can be provided
+            e.g. key=[value1&value2&value3]. Alternatively, ranges can also be
+            specified e.g. key=[value1:value3].
             When a range is specified, this is inclusive of the endpoints of
             the range.
         units (list):
             List of units as strings corresponding to each coordinate in the
             list of constraints. One or more "units" may be None and units may
             only be associated with coordinate constraints.
+        ignore_failure (bool):
+            Option to ignore constraint match failure and return the input
+            cube.
+            Default is false.
 
     Returns:
         iris.cube.Cube:
             A single cube matching the input constraints or None. If no
             sub-cube is found within the cube that matches the constraints.
     """
-    return extract_subcube(cube, constraints, units)
+    from improver.utilities.cube_extraction import extract_subcube
+
+    constraints = [item.replace('&', ', ') for item in constraints]
+    if units is not None:
+        units = [item.replace('&', ', ') for item in units]
+
+    result = extract_subcube(cube, constraints, units)
+
+    if result is None and ignore_failure:
+        return cube
+    elif result is None:
+        msg = "Constraint(s) could not be matched in input cube"
+        raise ValueError(msg)
+    return result
 
 
 if __name__ == '__main__':
-    main()
+    process()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [No value for argument 'constraints' in function call](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681435)